pkg/trace/config/apply.go

```diff
@@ -254,6 +254,9 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	c.StatsWriterConfig = readStatsWriterConfigYaml()
 	c.TraceWriterConfig = readTraceWriterConfigYaml()
 
+	// allow an extra 10% of the maximum number of connections for parallel writes to the Datadog API
+	c.TraceWriterConfig.SenderConfig.MaxConnections = c.ConnectionLimit / 10
```

Comment:
What's the deal with `c.ConnectionLimit` vs `c.MaxConnections`?

Also, shouldn't we probably do a `max(c.MaxConnections, c.ConnectionLimit / 10)` here (or at least `max(1, c.ConnectionLimit / 10)`?

Reply:
> What's the deal with c.ConnectionLimit vs c.MaxConnections?

Yeah, that's confusing. Sorry about that. `MaxConnections` is deprecated. I've completely removed it.

> Also, shouldn't we probably do a max(c.MaxConnections, c.ConnectionLimit / 10) here (or at least max(1, c.ConnectionLimit / 10)?

You're right, we should make sure we get at least 1 here. Done!